pkg/dns/aws/dns.go

```diff
@@ -244,7 +267,25 @@ func (m *Provider) getZoneID(zoneConfig configv1.DNSZone) (string, error) {
 
 	// Look up and cache the ID for these tags.
 	var id string
+	var err error
+	if m.tags != nil {
+		id, err = m.lookupZoneID(zoneConfig)
+	} else {
+		id, err = m.lookupZoneIDWithoutResourceTagging(zoneConfig)
+	}
+	if err != nil {
+		return id, err
+	}
+
+	// Update the cache
+	m.idsToTags[id] = zoneConfig.Tags
+	log.Info("found hosted zone using tags", "zone id", id, "tags", zoneConfig.Tags)
```

Comment:
This log statement should be updated, since we may have gotten here with or without tags, right?